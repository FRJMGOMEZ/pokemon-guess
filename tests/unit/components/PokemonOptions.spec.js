import { shallowMount } from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions';
import {pokemons} from '../mocks/pokemons.mock';

describe('PokemonOptions Component', () => {

    let wrapper;
    beforeEach(()=>{
        wrapper = shallowMount(PokemonOptions, { props: { pokemons } });
    })
    

    test('debe de hacer match con el snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    test('Debe de mostrar las 4 opciones correctamente',()=>{
        const liArr = wrapper.findAll('li');
        expect(liArr.length).toBe(4);
        liArr.forEach((_,index)=>{
             expect(liArr[index].text()).toBe(pokemons[index].name);
        });
    })

    

    test('debe de emitir "selection-pokemon" con sus respectivos parámetros al hacer click',()=>{
        const liArr= wrapper.findAll('li');
        liArr.forEach((_, index) => {
            liArr[index].trigger('click');

            /// cuantas veces se ha disparado el resgitro ///
            const emissionsRegister = wrapper.emitted('selection');
            expect(emissionsRegister.length).toBe(index + 1);

            //// valores de la emisión de evento ////
            const emissionValues = emissionsRegister[index];
            expect(emissionValues.length).toBe(1);

            ///// primer valor de emisión /////
            const emissionFirstValue = emissionValues[0];
            expect(emissionFirstValue).toBe(pokemons[index].id);
        });   
    })
})
 