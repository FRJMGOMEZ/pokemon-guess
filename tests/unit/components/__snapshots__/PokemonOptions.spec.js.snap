// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>Charizar</li>
    <li>Bulbasur</li>
    <li>Squirrel</li>
    <li>Picachu</li>
  </ul>
</div>
`;
